expyfun/stimuli/_tracker.py

```diff
@@ -0,0 +1,837 @@
+        The starting level of the tracker.
+    change_criteria : list of int | None
+        The points along the tracker to change its step sizes. Has an effect
+        where ``step_size_up`` and ``step_size_down`` are
+        lists. The length of ``change_criteria`` must be the same as the
+        length of ``step_size_up`` and ``step_size_down``. See note below for
+        more specific usage information. Should be None if static step sizes
+        are used.
+    change_rule : str
+        Whether to change parameters based on 'trials' or 'reversals'.
+    x_min : float
+        The minimum value that the tracker level (``x``) is allowed to take.
+    x_max : float
+        The maximum value that the tracker level (``x``) is allowed to take.
+
+    Returns
+    -------
+    tracker : instance of TrackerUD
+        The up-down tracker object.
+
+    Notes
+    -----
+    It is common to use dynamic parameters in an adaptive tracker. For example:
+    the step size is often large for the first couple reversals to get in the
+    right general area, and then it is reduced for the remainder of the track.
+    This class allows that functionality by defining that parameter with a list
+    of values rather than a scalar. The parameter will change to the next value
+    in the list whenever a change criterion (number of trials or reversals) is
+    encountered. This means that the length of the list defining a dynamic
+    parameter must always be the same as that of ``change_criteria``, and the
+    first element of change_criteria must always be 0.
+    For the example given above:
+        ``step_size_up=[1., 0.2]``, ``step_size_down=[1., 0.2]``,
+        ``change_criteria=[0, 2]``, ``change_rule='reversals'``
+    would change the step sizes from 1 to 0.2 after two reversals.
+
+    If static step sizes are used, both ``step_size_up``
+    and ``step_size_up`` must be scalars and ``change_criteria`` must be None.
+    """
+    def __init__(self, callback, up, down, step_size_up, step_size_down,
+                 stop_criterion, stop_rule, start_value, change_criteria=None,
+                 change_rule='reversals', x_min=None, x_max=None):
+        self._callback = _check_callback(callback)
+        self._up = up
+        self._down = down
+        self._stop_criterion = stop_criterion
+        self._stop_rule = stop_rule
+        self._start_value = start_value
+        self._x_min = x_min
+        self._x_max = x_max
+
+        if change_criteria is None:
+            change_criteria = [0]
+        elif change_criteria[0] != 0:
+            raise(ValueError('First element of change_criteria must be 0.'))
+        self._change_criteria = np.asarray(change_criteria)
+        if change_rule not in ['trials', 'reversals']:
+            raise(ValueError, "change_rule must be either 'trials' or "
+                  "'reversals'")
+        self._change_rule = change_rule
+
+        step_size_up = list(np.atleast_1d(step_size_up))
```

Comment:
wondering why you're casting to a list, when a few lines later you cast back to array.  Also, want to call your attention to what might happen if the user gives a 2D array for `step_size_up`...  as currently implemented, I think it will still pass these tests as long as its first dimension is the right length.  Presumably it will error out later, but might be worth catching it here?